### ISSUE TITLE
Mac CI/CD fails because of Java 8

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/sarif4k#119]
https://github.com/actions/setup-java/issues/625 -> https://github.com/actions/setup-java/issues/625#issuecomment-2096064399

Probably caused by: https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/

Probable solution:
```patch
-macos-latest
+macos-13 # Java 8 is not supported on macos-14+ (M1).
```

I saw now new issue related to this at https://github.com/actions/runner-images/issues

**Comment by TWiStErRob** [comment]
@detekt/maintainers thoughts?

**Comment by 3flex** [comment]
Downgrade to macros 13 for now I think.

11 is still supported by GitHub runners (though deprecated), so there's quite some time until 13 is dropped.

Alternatively move to a Java distribution that supports Java 8 on aarch64 (Coretto or Zulu for example).

**Comment by TWiStErRob** [comment]
Merged a fix to unblock the dependency updates, we can change it if the solution is not the best.

3flex's comment made me realise that we should stay on latest of machine because it should be faster during compilation even if it needs to download a Java.
